src/hawkmoth/__init__.py

```diff
@@ -39,6 +40,31 @@ class _AutoBaseDirective(SphinxDirective):
     _domain: Optional[str] = None
     _docstring_types: Optional[list[type[docstring.Docstring]]] = None
 
+    def __init__(self, name, arguments, options,
+                 content, lineno, content_offset,
+                 block_text, state, state_machine):
+
+        super().__init__(name, arguments, options,
+                         content, lineno, content_offset,
+                         block_text, state, state_machine)
+
+        cpath = self.env.config.hawkmoth_compiler
+        autoconf = self.env.config.hawkmoth_autoconf
+
+        ignored_options = [x for x in autoconf if x not in ['stdinc']]
+        if len(ignored_options) > 0:
+            self.logger.warning(f'autoconf: {ignored_options} unsupported option(s) ignored')
+
+        self._clang_args_post = []
+        if 'stdinc' in autoconf:
+            if cpath:
+                if self._domain == 'c':
+                    self._clang_args_post = compiler.get_include_args(cpath, 'c')
+                else:
+                    self._clang_args_post = compiler.get_include_args(cpath, 'c++')
+            else:
+                self.logger.warning('autoconf: \'stdinc\' option ignored (missing compiler)')
+
```

Comment:
This still gets called once per each directive used in the documentation. There's one instance for directive.

The hard part is that `setup()` can't use the config options. IIUC you'd have to use a Sphinx event to do it once.

Reply:
The alternative is to simplify and go back to the first version (instead of the constructor) and optimize later.

Reply:
Ah, of course. Well, I did this on a crowded flight, it was bound to happen :sweat_smile: 

I'll look into it when I can. There are other (maybe less clean) alternatives like caching the result if we go for a temporary solution, but the events should work well enough if there's a suitable one.